amphimixis/general/general.py

```diff
@@ -88,88 +84,20 @@ class IBuildSystem(ABC):
-
-        path_ = ""
-        for root, _, files in walk(project.path):
-            for name in files:
-                if name == "CMakeLists.txt":
-                    path_ = path.join(root, name)
-                    return path_
-        return path_
+    def get_runner_prompt(project: Project, build: Build) -> str:
+        """Generate runner prompt"""
 
     @staticmethod
-    def insert_config_flags(project: Project, build: Build, command: str) -> str:
-        """Method insert flags in 'command' in line with call of build system
-        or return string with command which run build system with inserted flags
-        If 'command' is empty then return string in the format '${BuildSystem} ${config_flags}'
-        else return string 'command' with 'config_flags' inserted"""
-
-        if command != "":
-            raise NotImplementedError
-
-        command = "cmake " + CMake.find_cmakelists_path(project) + " "
-        command += build.config_flags
-        command += " CXXFLAGS='" + build.compiler_flags + "'"
-        command += " CFLAGS='" + build.compiler_flags + "'"
-        return command
-
-    @staticmethod
-    def insert_runner_flags(project: Project, build: Build, command: str) -> str:
-        """Method insert flags in 'command' in line with call of runner
-        or return string with command which run runner with inserted flags
-        If 'command' is empty then return string in the format '${BuildSystem} ${runner_flags}'
-        else return string 'command' with 'runner_flags' inserted"""
-
-        raise NotImplementedError
-
-
-class Make(IBuildSystem):
-    """The Make implementation of IBuildSystem"""
-
-    @staticmethod
-    def insert_config_flags(project: Project, build: Build, command: str) -> str:
-        """Method insert flags in 'command' in line with call of build system
-        or return string with command which run build system with inserted flags
-        If 'command' is empty then return string in the format '${BuildSystem} ${config_flags}'
-        else return string 'command' with 'config_flags' inserted"""
-
-        raise NotImplementedError
+    @abstractmethod
+    def _insert_toolchain(build: Build) -> str:
+        """Return flag that specify toolchain in build system call"""
 
     @staticmethod
-    def insert_runner_flags(project: Project, build: Build, command: str) -> str:
-        """Method insert flags in 'command' in line with call of runner
-        or return string with command which run runner with inserted flags
-        If 'command' is empty then return string in the format '${BuildSystem} ${runner_flags}'
-        else return string 'command' with 'runner_flags' inserted"""
-
-        if command != "":
-            raise NotImplementedError
-
-        return "make -j" + str(cpu_count())
-
-
-build_systems_dict: dict[str, type[IBuildSystem]] = {
-    "cmake": CMake,
-    "make": Make,
-}
+    @abstractmethod
+    def _insert_sysroot(build: Build) -> str:
```

Comment:
```suggestion
    def insert_sysroot(build: Build) -> str:
```
an interface is a public API, it shouldn't have private methods